resources/js/components/pages/SettingsChangeEmail.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+    <div>
+        <div v-if="enableReset" class="flex flex-col space-y-4">
+            <Form
+                id="changeEmailForm"
+                ref="formRef"
+                class="space-y-6"
+                :validation-schema="validation"
+                @submit="changeEmail"
+            >
+                <FormInput
+                    v-model="currentEmail"
+                    label="Current Email"
+                    class="sm:w-[430px] flex-1 sm:flex-none text-gray-400"
+                    name="currentEmail"
+                    disabled
+                />
+                <FormInput
+                    v-model="newEmail"
+                    label="New Email"
+                    placeholder="New Email"
+                    class="sm:w-[430px] flex-1 sm:flex-none text-gray-400"
+                    name="newEmail"
+                />
+                <div class="flex space-x-4">
+                    <Btn @click="enableReset = false" id="changeEmailCancelBtn">Cancel</Btn>
+                    <Btn is-submit primary :loading="isLoading" id="changeEmailSubmitBtn">Change Email</Btn>
+                </div>
+            </Form>
+        </div>
+        <div v-else>
+            <Btn primary @click="enableReset = true" dusk="changeEmailBtn">Change Email</Btn>
+        </div>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue';
+import FormInput from '../FormInput.vue';
+import { AuthApi } from '~/api/auth';
+import Btn from '../Btn.vue';
+import * as yup from 'yup';
+import { snackbarErrors } from '~/util';
+import snackbar from '~/util/snackbar';
+import { Form } from 'vee-validate';
+import { useAppStore } from '~/store';
+import { ApiService } from '~/api';
+
+const appStore = useAppStore();
+
+const currentEmail = ref(appStore.user?.email);
+const newEmail = ref();
+const isLoading = ref(false);
+const formRef = ref();
+const enableReset = ref(false);
+
+const validation = yup.object().shape({
+    newEmail: yup.string().email().required(),
+});
+
+const isValid = async () => {
+    await formRef.value.validate();
+    return formRef.value.getMeta().valid;
+};
+
+const changeEmail = async () => {
+    if (!(await isValid())) return;
+
+    isLoading.value = true;
+    try {
+        const res = await AuthApi.updateUser(newEmail.value);
+        if (res.data.UpdateUser) {
+            snackbar.success({
+                title: 'Email changed',
+                text: 'Your email has been changed successfully.',
+            });
+            appStore.logout();
```

Comment:
Need to refresh the csrf token here, else FE will get invalid csrf.

Reply:
@enjinabner the logout method has the reloadCSRF call